meteor/client/ui/Shelf/BucketPanel.tsx

```diff
@@ -737,11 +751,34 @@ export const BucketPanel = translateWithTracker<Translated<IBucketPanelProps>, I
 						}
 					}
 				}
+				private adLibIsDisabled = (adlib: BucketAdLibItem) => {
+					return (
+						adlib.showStyleBaseId !== this.props.showStyleBaseId ||
+						(!!adlib.showStyleVariantId && adlib.showStyleVariantId !== this.props.showStyleVariantId)
+					)
+				}
 
 				render() {
 					const { connectDragSource, connectDragPreview, connectDropTarget } = this.props
 
 					if (this.props.showStyleBase) {
+						// Hide duplicates;
+						// only the first adLib found with a given uniquenessId will be displayed
+						const uniqueAdlibs = new Map<string, BucketAdLibItem>()
+						for (const adLibPiece of this.state.adLibPieces) {
+							const uniquenessId = adLibPiece.uniquenessId || unprotectString(adLibPiece._id)
```

Comment:
Is there a reason for using the uniquenessId ere instead of the externalId?

As I see it, in the backend we are using the externalId to figure out if they are the same, then in the frontend we are using the uniquenessId, with no clearly defined relationship between the properties.

To me it would make sense for two passes, first to filter out duplicate externalIds, then to filter out duplicate uniquenessIds.

uniquenessId was originally added so that adlibs from different parts which are identical could be identified and hidden. But this pr 'claims' uniquenessId as a core property for bucket adlibs, even though externalId should be functionally identical